### PR TITLE
chore: bump upstream v0.46.13

### LINKED ISSUE #310: Pull upstream changes from v0.46.13
## Context

https://github.com/cosmos/cosmos-sdk/releases/tag/v0.46.12

## Proposal

Since we're using the Cosmos SDK v0.46.x release branch for our first mainnet release, we should upgrade to the latest patch release prior to mainnet

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

pulls the latest from upstream, which include bareberry patch. also bumps core to v0.34.28

closes #310 

ready for review, but blocked by https://github.com/celestiaorg/celestia-core/pull/1021

### CODE REVIEW

**Comment by evan-forbes** [comment]
called `make proto-check-deps` which downloaded the most recent stuff, which caused the proto breakage

**Comment by evan-forbes** [comment]
hmm apparently we can't sync `mocha-2` using this because of 

```
7:29AM ERR Rejected proposal block: err="failed to load state at height 0; no commit info found (latest height: 0): invalid request" proposer="�6�\v��0®����e\x14\x02��E�" reason="failure to load query context"
```

looking into this now, but converting to a draft

**Review by cmwaters — APPROVED** [review]

File: proto/tendermint/types/types.proto [hunk 1]
```diff
@@ -87,68 +87,6 @@ message Data {
...
-// coded version of the data in Block.Data.
-// Therefor the original Block.Data is arranged in a
-// k × k matrix, which is then "extended" to a
-// 2k × 2k matrix applying multiple times Reed-Solomon encoding.
-// For details see Section 5.2: https://arxiv.org/abs/1809.09044
-// or the Celestia specification:
-// https://github.com/celestiaorg/celestia-specs/blob/master/specs/data_structures.md#availabledataheader
-// Note that currently we list row and column roots in separate fields
-// (different from the spec).
-message DataAvailabilityHeader {
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply proto/tendermint/types/types.proto:147]: we just don't need this period for the sdk. I also believe its in the app now

**Comment by evan-forbes** [comment]
> We haven't added prepareProposalState and processProposalState yet right?
I assume this comes from running process proposal on the first block. In our CometBFT fork we run ProcessProposal in blocksync however mainline CometBFT doesn't actually do this.

#326 addresses a portion of this by replicating what is done in upstream v0.47.x, which is branching the state directly from the commit multi store to create a context which is passed to the new ABCI methods. However, this still doesn't fix being able to access that state at height 1, so we still have the same issues as we had before :slightly_frowning_face: https://github.com/celestiaorg/celestia-app/issues/1899

I'm still unsure of how upstream handles this with the baseapp refactor, since as far as state access goes for the Prepare/Process #326 is doing the exact same thing as upstream

**Comment by evan-forbes** [comment]
> But it looks like its trying to access the state at height 0.
Also as I said, in blocksync we are calling ProcessProposal but in CometBFT they don't call it (i.e. it might not be SDK but at the Tendermint level)

yeah I remember that, the 

> 7:29AM ERR Rejected proposal block: err="failed to load state at height 0;

part was caused by us not passing the correct height (1) during `ProcessProposal`, which is fixed by #326 since we're passing the header instead of relying on the header included in other baseapp state.

However, we're still having issues in `Prepare/ProcessProposal` for accessing state at height 1, which upstream has solved.

**Review by rootulp — APPROVED** [review]

**Pull request merged**